tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -199,6 +199,83 @@ static PyObject * THPVariable_range(PyObject* self, PyObject* args, PyObject* kw
   END_HANDLE_TH_ERRORS
 }
 
+inline Tensor dispatch_full(
+    IntArrayRef size,
+    Scalar fill_val,
+    const TensorOptions& options) {
+  torch::utils::maybe_initialize_cuda(options);
+  pybind11::gil_scoped_release no_gil;
+  return at::full(size, fill_val, options);
+}
+
+inline Tensor dispatch_full(
+    IntArrayRef size,
+    Scalar fill_val,
+    c10::optional<DimnameList> names,
+    const TensorOptions& options) {
+  torch::utils::maybe_initialize_cuda(options);
+  pybind11::gil_scoped_release no_gil;
+  return at::full(size, fill_val, names, options);
+}
+
+inline Tensor dispatch_full(
+    IntArrayRef size,
+    Scalar fill_val,
+    Tensor result) {
+  pybind11::gil_scoped_release no_gil;
+  return at::full_out(result, size, fill_val);
+}
+
+static PyObject * THPVariable_full(PyObject* self, PyObject* args, PyObject* kwargs) {
```

Comment:
So this handwritten binding is based on the one for `arange`, which is based on an older version of the codegen. It works fine except that it's missing the `torch_function` hook functionality that's now present in the generated bindings. If you could add that to `full` and also `arange`, that would be awesome. [Here's the current codegen for `full` that includes the `has_torch_function` intercept.](https://our.internmc.facebook.com/intern/paste/P127492935/)

Reply:
Added the check to arange and randint, too, but no the deprecated int. I verified that those functions and torch.full appear on the list of overridable functions, too.